### ISSUE TITLE
Extremely high loss value for each epoch, not increasing in accuracy.

### ISSUE THREAD

**Issue by jeshaitan** [fchollet/keras#1987]
```
from keras.models import Sequential
from keras.layers.core import Dense, Activation, Masking
from keras.layers.recurrent import LSTM

#build and train model
in_dimension = 3
out_dimension = 2

model = Sequential()
model.add(Masking([[0,0,0]], input_shape=(max_sequence_length, in_dimension)))
model.add(LSTM(output_dim=out_dimension, input_dim=in_dimension, return_sequences=False))
model.add(Activation("softmax"))
model.add(Dense(output_dim=out_dimension, input_dim=in_dimension))
model.add(Activation("softmax"))
model.compile(loss="categorical_crossentropy", optimizer="rmsprop")

model.fit(padded_training_seqs, training_final_steps, nb_epoch=5, batch_size=1)
```

`padded_training_seqs` is an an array of sequences of [latitude, longitude, temperature], all padded to the same length with values of [0,0,0]. When I train this network, each epoch gives me a loss of about 63, and only decreases to about 62 after 100 epochs. 
This is causing a `model.predict` call later in the code to give values that are completely off of the training values. For example, most of the training values in each sequence is around `[40, 40, 20]`, but the RNN outputs values consistently around `[0.4, 0.5]`, which causes me to think something is wrong with the masking layer. 

The training X (`padded_training_seqs`) data looks something like this (only much larger): 

```
[
[[43.103, 27.092, 19.078], [43.496, 26.746, 19.198], [43.487, 27.363, 19.092], [44.107, 27.779, 18.487], [44.529, 27.888, 17.768]], 
[[44.538, 27.901, 17.756], [44.663, 28.073, 17.524], [44.623, 27.83, 17.401], [44.68, 28.034, 17.601], [0,0,0]],
[[47.236, 31.43, 13.905], [47.378, 31.148, 13.562], [0,0,0], [0,0,0], [0,0,0]]
]
```

and the training Y (`training_final_steps`) data looks like this:

```
[
[44.652, 39.649], [37.362, 54.106], [37.115, 57.66501]
]
```

**Comment by keunwoochoi** [comment]
Have you normalised the data (Zero-mean and std of 1)? I'd check by putting `BatchNormalization(axis=?, input_shape=(your input shape))` at the beginning of the model.
